pkg/envoy/ads/stream.go

```diff
@@ -332,6 +332,31 @@ func isCNforProxy(proxy *envoy.Proxy, cn certificate.CommonName) bool {
 	return identityForCN == proxy.Identity.ToK8sServiceAccount()
 }
 
+func getCertificateCommonNameMeta(cn certificate.CommonName) (envoy.ProxyKind, uuid.UUID, identity.ServiceIdentity, error) {
+	// XDS cert CN is of the form <proxy-UUID>.<kind>.<proxy-identity>.<namespace>.<trust-domain>
```

Comment:
Is the proxy-identity the service account? For consistency could this be updated to service account. The form of the XDS cert CN is documented elsewhere as <ProxyUUID>.<kind>.<serviceAccount>.<namespace>.<trust-domain

Reply:
We should be hiding K8s abstractions as much as possible, so I think it makes more sense to keep it as identitiy. I'll fix the comment though.